src/VisualStudio/Core/Next/Remote/JsonRpcClient.cs

```diff
@@ -18,68 +18,37 @@ namespace Microsoft.VisualStudio.LanguageServices.Remote
     internal class JsonRpcClient : IDisposable
     {
         private readonly JsonRpc _rpc;
-        private readonly CancellationToken _cancellationToken;
 
-        public JsonRpcClient(
-            Stream stream, object callbackTarget, bool useThisAsCallback, CancellationToken cancellationToken)
+        public JsonRpcClient(Stream stream, object callbackTarget, bool useThisAsCallback)
         {
             Contract.Requires(stream != null);
 
             var target = useThisAsCallback ? this : callbackTarget;
-            _cancellationToken = cancellationToken;
 
             _rpc = new JsonRpc(stream, stream, target);
             _rpc.JsonSerializer.Converters.Add(AggregateJsonConverter.Instance);
 
             _rpc.Disconnected += OnDisconnected;
         }
 
-        public async Task InvokeAsync(string targetName, params object[] arguments)
+        public Task InvokeWithCancellationAsync(string targetName, object[] arguments, CancellationToken cancellationToken)
         {
-            _cancellationToken.ThrowIfCancellationRequested();
-
-            try
-            {
-                await _rpc.InvokeAsync(targetName, arguments).ConfigureAwait(false);
-            }
-            catch (ObjectDisposedException)
-            {
-                // object disposed exception can be thrown from StreamJsonRpc if JsonRpc is disposed in the middle of read/write.
-                // the way we added cancellation support to the JsonRpc which doesn't support cancellation natively
-                // can cause this exception to happen. newer version supports cancellation token natively, but
-                // we can't use it now, so we will catch object disposed exception and check cancellation token
-                _cancellationToken.ThrowIfCancellationRequested();
-                throw;
-            }
+            return _rpc.InvokeWithCancellationAsync(targetName, arguments, cancellationToken);
```

Comment:
ah, I see. but the _rpc we can use the new name, but can we just keep InvokeAsync in ourside? since we are not planning to provide both InvokeAsync and InvokeWithCancellationAsync.

we using StreamJson as underlying lib is implementation detail, and one that sit on top of us, doesn't need to know that we are using StreamJson.